src/JsonSchema/Validator.php

```diff
@@ -43,13 +41,50 @@ public function check($value, $schema = null, JsonPointer $path = null, $i = nul
      * types defined in the schema whenever possible. Note that the first argument is passed by reference,
      * so you must pass in a variable.
      *
+     * This method is now a simple alias, and exists only so that the API is not fundamentally broken for
+     * people who use this. This method should probably be officially deprecated at some point.
+     *
      * {@inheritDoc}
      */
-    public function coerce(&$value, $schema = null, JsonPointer $path = null, $i = null)
+    public function coerce(&$value, $schema = null)
     {
-        $validator = $this->factory->createInstanceFor('schema');
-        $validator->coerce($value, $schema);
+        // Preserve the current checkMode, because users of this API expect to be
+        // able to set it via the factory before calling this method.
+        $checkMode = $this->factory->getCheckMode() | self::CHECK_MODE_COERCE;
 
-        $this->addErrors(array_unique($validator->getErrors(), SORT_REGULAR));
+        return $this->validate($value, $schema, $checkMode);
+    }
+
+    /**
+     * Validates the given data against the schema and returns an object containing the results
+     * Both the php object and the schema are supposed to be a result of a json_decode call.
+     * The validation works as defined by the schema proposal in http://json-schema.org.
+     *
+     * @param mixed $value Decoded JSON document to validate
+     * @param mixed $schema Decoded JSON schema to validate against
+     * @param int $checkMode Set various options to change the validation behavior
+     *
+     * {@inheritDoc}
+     */
+    public function validate(&$value, $schema = null, $checkMode = self::CHECK_MODE_NORMAL)
```

Comment:
I like the idea that you have sort of an advanced usage method where you can explicitly state the kinds of bells and whistles you want. I would argue that this should be the *main* entry point, and that everything else (`coerce`, `check`, `forceDefaults`, etc) could be considered helpers, shortcuts, or possibly even eventually deprecated and removed entirely.

One suggestion you might consider is to convert all instances of `_check` (which is confusingly public and mirroroing `check`) with `validate`, and just pass $checkmode along everywhere.